GitUI/Avatars/InitialsAvatarProvider.cs

```diff
@@ -65,35 +87,72 @@ private static (string? name, char[]? separator) NameSelector(string? name, stri
                 return null;
             }
 
+            string name = names[0];
+
             // If only a single valid name-element is found ...
             if (names.Length == 1)
             {
                 // ... and that name-element is only a single character long ...
-                if (names[0].Length == 1)
+                if (name.Length == 1)
                 {
                     // ... return that character as uppercase
-                    return names[0][0].ToString().ToUpper();
+                    return name[0].ToString().ToUpper();
+                }
+
+                if (char.IsUpper(name[1]))
+                {
+                    return $"{char.ToUpper(name[0])}{name[1]}";
+                }
+
+                string[] splitNames = name.Split(_emailInitialSeparator);
+
+                if (splitNames.Length > 1)
+                {
+                    return GetInitialsFromNames(splitNames);
+                }
+
+                char[] upperChars = name.Where(char.IsUpper).ToArray();
+                if (upperChars.Length > 1)
+                {
+                    return $"{upperChars[0]}{upperChars[upperChars.Length - 1]}";
                 }
 
                 // return first letter upper-case and second letter original/lower case.
-                return $"{char.ToUpper(names[0][0])}{names[0][1]}";
+                return $"{char.ToUpper(name[0])}{name[1]}";
             }
 
             // Return initials from first and last name-element as uppercase
-            return $"{names[0][0]}{names[names.Length - 1][0]}".ToUpper();
+            return $"{name[0]}{names[names.Length - 1][0]}".ToUpper();
         }
 
         private readonly Graphics _graphics = Graphics.FromImage(new Bitmap(1, 1));
         private readonly Brush _textBrush = new SolidBrush(Color.WhiteSmoke);
 
-        private readonly Color[] _avatarColors =
+        private readonly Color[] _avatarColors = AppSettings.AvatarAuthorInitialsPalette.Split(",").Select(ConvertToColor).WhereNotNull().Cast<Color>().ToArray();
+
+        private static Color? ConvertToColor(string c)
         {
-            Color.RoyalBlue,
-            Color.DarkRed,
-            Color.Purple,
-            Color.ForestGreen,
-            Color.DarkOrange
-        };
+            if (!c.StartsWith("#"))
+            {
+                try
+                {
+                    return Color.FromName(c);
+                }
+                catch (Exception)
+                {
+                    return Color.Black;
+                }
+            }
+
+            try
+            {
+                return ColorTranslator.FromHtml(c);
+            }
+            catch
+            {
+                return Color.Black;
+            }
```

Comment:
Don't make it so complicated...

![image](https://user-images.githubusercontent.com/3629489/229351373-09e84137-4cba-4b2c-963d-4d7babd1a22d.png)
![image](https://user-images.githubusercontent.com/3629489/229351391-5c08030e-ac45-40ef-828f-b3d979184199.png)
![image](https://user-images.githubusercontent.com/3629489/229351424-128f8977-1f39-45f3-bab2-4b8b5bf389eb.png)


```C#
Color named = Color.Cyan;
Color other = Color.FromArgb(named.ToArgb() + 1);

var translations = new { Named = named, NamedTranslated = ColorTranslator.ToHtml(named), Other = other, OtherTranslated = ColorTranslator.ToHtml(other) };

translations.Dump("TranslatedTo");

var BackToColors = new { Named = ColorTranslator.FromHtml(translations.NamedTranslated), Other = ColorTranslator.FromHtml(translations.OtherTranslated) };
BackToColors.Dump("RoundTrip");

```

Reply:
```suggestion
             try
            {
                return ColorTranslator.FromHtml(c);
            }
            catch
            {
                return Color.Black;
            }
```

Reply:
```C#
Dictionary<string, Color> colorDict = new Dictionary<string, Color>();
foreach (var prop in typeof(Color).GetProperties())
{
	if (prop.PropertyType == typeof(Color))
	{
		Color color = (Color)prop.GetValue(null);
		colorDict.Add(prop.Name, color);
	}
}

colorDict.Keys.Select(k => System.Drawing.ColorTranslator.FromHtml(k)).Dump();
```

Did not cause an exception  while 

![image](https://user-images.githubusercontent.com/3629489/229352714-903437a2-f8c7-4324-a3e0-8ec50d0aea61.png)

Reply:
I pushed changes hours ago. It seems that you are still basing your comments on the old version. Have a look...

Reply:
Not needed

Stop using fromname

![image](https://user-images.githubusercontent.com/3629489/229355430-96f05732-ae02-4f1f-995c-41262edadebf.png)
As I have proven in my comments all you need is the colortranslator.

Reply:
https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/ColorConverter.cs,109

https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Advanced/ColorTranslator.cs,228

See how ColorTranslator handles all the names and even the LightGray name difference?

Reply:
.net versions of links 

https://source.dot.net/#System.Drawing.Primitives/System/Drawing/Color.cs,475

Colors used by the abov emethod https://source.dot.net/System.Drawing.Primitives/src/libraries/Common/src/System/Drawing/ColorTable.cs.html#6532ea3b4c32baf5

While Translator is the following

https://source.dot.net/#System.Drawing.Primitives/System/Drawing/ColorTranslator.cs,218

Reply:
Indeed. Done.